src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -142,9 +142,9 @@ bool IsValidTriggerCharacterForDocument(Document document, char triggerCharacter
                 var completionItem = new TCompletionItem
                 {
                     Label = completeDisplayText,
-                    InsertText = item.Properties.ContainsKey("InsertionText") ? item.Properties["InsertionText"] : completeDisplayText,
-                    SortText = item.SortText,
-                    FilterText = item.FilterText,
+                    InsertText = (item.Properties.ContainsKey("InsertionText") && !item.Properties["InsertionText"].Equals(item.DisplayText, StringComparison.Ordinal)) ? item.Properties["InsertionText"] : null,
```

Comment:
can you extract this out below now that the logic is more complex?  Similar to the commitcharacters, only specify the insert text if it's different from the display text.

And I believe this should be checking against the Label (completeDisplayText) since we should only de-dupe when the label is the same (as the label will be used to insert)

Reply:
for an example, consider an intellicode suggestion which is prefixed by a ⭐ in the display of the completion list.  But we don't want to actually insert the ⭐.  The ⭐should be part of the DisplayTextPrefix, so if we only compare against the DisplayText, we would accidentally insert ⭐.